src/xmpp/dns.go

```diff
@@ -27,7 +27,18 @@ func HomeServerAddrs(jid JID) (addr []string, err error) {
 
 	// Build list of "host:port" strings.
 	for _, a := range addrs {
-		addr = append(addr, fmt.Sprintf("%s:%d", a.Target, a.Port))
+		target := parseTargetDomainName(a.Target)
+		addr = append(addr, fmt.Sprintf("%s:%d", target, a.Port))
+	}
+	return
+}
+
+// Remove the last dot in the domain name if exist
+func parseTargetDomainName(domainName string) (ret string) {
+	if domainName[len(domainName)-1] == '.' {
```

Comment:
Out of interest, why is this necessary? But anyway, what about just `domainName = strings.TrimRight(domainName, ".")` instead of the slice and recursive call?